### PR TITLE
C51-34: Add BEM classes to HTML and move the title inside panel header and misc fixes

### PR DESCRIPTION
**Pull request by swastikpareek**
## Overview
This PR covers UI enhacements for sticky header and basic HTML fixes for fitler panel on case list

## Before/After

* For the HTML fixes

### Before
c51-34 - before

### After

c51-34 - after

* Sticky header fixes 

### Before
c51-39 - header glitch before

### After
c51-39 - header glitch after

## Technical Details
 * This PR fixes sticky header scroll glitch by adding top padding to the table element when the header sticks to the top by hooking in `affixed-top.bs.affix` event and add logic
 * Update Unit tests
 * Fix the civicaseSearch directive. 
 ** Changed selector from just attribute to all. 
 ** Add `replace: true` to cut down the extra wrapper divs
 ** Renamed `civicase-search` attribute to `filter` attrbute to increase readability.
 * Add BEM classes to the HTML. Classes used
 ** **civicase__case-filter-panel** : For search panel container
 ** **civicase__case-filter-panel .panel-header** : For search panel header
 ** **civicase__case-filter-panel__title** : For search panel title
 ** **civicase__case-filter-container**: For search panel filters container
 ** **civicase__case-filter__input**: For search panel filter dropdowns
 ** **civicase__case-filter-panel__button**: For search panel filter buttons
 *** **civicase__case-filter-panel__button--search**: For search panel search filter buttons
 *** **civicase__case-filter-panel__button--other**: For search panel other criterion filter buttons
 *** **civicase__case-filter-panel__button--clear**: For search panel clear filter buttons

## Other details
* All the HTML classes are not fixed as per BEM as these will be taken under styling ticket which is C51-36
* For now the third dropdown is added as a copy of second filter dropdown so this could be themed properly in the style C51-36 ticket. The functionality will be fixed as a part of C51-35 ticket.

### CODE REVIEW

File: ang/civicase/CaseListDirectives.js [hunk 1]
```diff
@@ -64,6 +64,11 @@
               .on('affixed.bs.affix', function () {
                 $header.scrollLeft($table.scrollLeft());
                 $header.css('top', bodyPadding + 'px');
+                $table.css('padding-top', $header.height() + 'px');
```
Comment (deb1990) [inline ang/civicase/CaseListDirectives.js]: Please add a comment in the code explaining why this is necessary.

File: ang/civicase/CaseListDirectives.js [hunk 2]
```diff
@@ -64,6 +64,11 @@
...
+              })
+              .on('affixed-top.bs.affix', function () {
+                $header.css('top', 0);
```
Comment (deb1990) [inline ang/civicase/CaseListDirectives.js]: Please add a comment in the code explaining why this is necessary.

File: ang/civicase/CaseList.html [hunk 3]
```diff
@@ -1,15 +1,14 @@
 <div crm-ui-debug="cases"></div>
 <div crm-ui-debug="viewingCase"></div>
 <div crm-ui-debug="viewingCaseTab"></div>
-<h1 crm-page-title>{{ pageTitle }}</h1>
 <div id="bootstrap-theme" class="clearfix civicase__container">
-
-  <div class="panel panel-default civicase-search-panel clearfix" ng-class="{'case-is-focused': caseIsFocused}" civicase-search="filters" hidden-filters="hiddenFilters" expanded="searchIsOpen" on-search="applyAdvSearch(selectedFilters)">
+  <h1 crm-page-title>Manage Cases</h1>  <!-- Added for SEO purpose (This will be hidden through css) -->
```
Comment (deb1990) [inline ang/civicase/CaseList.html]: I can see this element. Its not hidden yet.

**Comment by deb1990** [comment]
@swastikpareek the BEM class name `civicase__filter-search-panel` does not say that its for case. What if we have more search/filters. We need to specify that its for `case-filter`. Overall these are my suggestions

```
** civicase__case-filter-panel                -  civicase__filter-search-panel
** civicase__case-filter-panel .panel-header  -  civicase__filter-search-panel__header
** civicase__case-filter-panel__title         -  civicase__filter-search-panel__header-title
** civicase__case-filters-container           -  civicase__filter-search-panel__filters-container
** civicase__case-filter__input               -  civicase__filter-search-panel__filter
** do we need this class?                     -  civicase__filter-search-panel__button
*** civicase__case-filter__submit             -  civicase__filter-search-panel__button--search
*** civicase__case-filter__toggle-view        -  civicase__filter-search-panel__button--other
*** civicase__case-filter__clear              -  civicase__filter-search-panel__button--clear
```

let me know if you want to discuss.

File: ang/civicase/CaseList.html (hunk 3 above)
Comment (swastikpareek) [reply ang/civicase/CaseList.html]: This will be styled (made hidden) in the next styling ticket C51-36.
Adding this in the PR description.

**Comment by swastikpareek** [comment]
@deb1990 : Updated BEM classes as per the information provided by you. However there will be buttons with a set of same styles (like line heights,color,font size etc) with some different states. This will be fixed eventually while styling it (as at that time it will be more clear which class can be used and which is unnecessary).

**Pull request merged**